File: packages/typescript/src/services/account.ts
import { BaseService } from './base.js';

export interface UserInfo {
  id: string;
  email: string;
  name?: string;
  plan: string;
  createdAt?: string;
}

export interface UserBudget {
  creditsUsed: number;
  creditsLimit: number;
  creditsRemaining: number;
  resetDate?: string;
}

export interface UserUsage {
  month: string;
  totalRequests: number;
  totalTokens: number;
  breakdown: Record<string, unknown>;
}

export class AccountService extends BaseService {

  async health(): Promise<Record<string, unknown>> {
    return this.request('GET', '/health');
  }

  async info(): Promise<UserInfo> {
    return this.request<UserInfo>('GET', '/v1/user/info');
  }

  async budget(): Promise<UserBudget> {
    return this.request<UserBudget>('GET', '/v1/user/budget');
  }

  async usage(): Promise<UserUsage> {
    return this.request<UserUsage>('GET', '/v1/user/stats');
  }
}

File: packages/python/scalix_sdk/services/account.py
"""Account service — user info, budget, usage."""

from __future__ import annotations

from typing import Any

from scalix_sdk.services.base import BaseService


class AccountService(BaseService):

    async def health(self) -> dict[str, Any]:
        return await self._request("GET", "/health")

    async def info(self) -> dict[str, Any]:
        return await self._request("GET", "/v1/user/info")

    async def budget(self) -> dict[str, Any]:
        return await self._request("GET", "/v1/user/budget")

    async def usage(self) -> dict[str, Any]:
        return await self._request("GET", "/v1/user/stats")
